### ISSUE TITLE
luminance-front cannot compile on arm64 or other target archs not hardcoded

### ISSUE THREAD

**Issue by kpreid** [phaazon/luminance-rs#509]
The way `luminance-front/src/lib.rs` is written,

```rust
#[cfg(all(any(target_arch = "x86", target_arch = "x86_64"), feature = "gl33"))]
pub type Backend = luminance_gl::GL33;

#[cfg(all(target_arch = "wasm32", feature = "webgl2"))]
pub type Backend = luminance_webgl::webgl2::WebGL2;
```

if the `target_arch` is not any of the explicitly listed ones, then no combination of feature requests will allow `luminance-front` to successfully compile. The `Cargo.toml` also has a similar obstacle, because it defines dependencies in terms of archs and not features. It seems like it would be possible to avoid this problem by making the conditions more like

```
cfg(any(feature = "webgl2", all(feature = "autoselect", target_arch = "wasm32")))
```

so that a requested feature can completely override the automatic selection. Alternatively, the gl33 backend could be selected by `not(target_arch = "wasm32")`. That wouldn't allow overriding the selection, but it would allow compilation with autoselection.

I've confirmed that with modifications to enable it, luminance-front, luminance-gl, and luminance-glfw will in fact compile on arm64, though I hit a runtime shader compilation error needing further investigation so I haven't verified full functionality. This came up while testing on a Chrome OS machine.

**Comment by phaazon** [comment]
Yeah that’s annoying, I thought about it lately. Fortunately very easy to fix, though. :) I will update that asap.
